autorosters/autorosters_main.py

```diff
@@ -0,0 +1,270 @@
+        self.process_matchschedule_data(matchschedule_data)
+        self.process_scoreboard_data(scoreboard_data)
+
+    def get_matchschedule_data(self):
+        matchschedule_data = self.site.cargo_client.query(
+            tables="MatchSchedule=MS, MatchScheduleGame=MSG",
+            fields=["MS.MatchId", "MSG.GameId", "MS.FF=MSFF", "MSG.FF=MSGFF", "MS.BestOf", "MS.Team1Final",
+                    "MS.Team2Final", "MS.Team1", "MS.Team2"],
+            join_on="MS.MatchId=MSG.MatchId",
+            where=f"MS.OverviewPage = '{self.overview_page}'",
+            order_by="MS.N_Page, MS.N_MatchInPage, MSG.N_GameInMatch"
+        )
+        return matchschedule_data
+
+    @staticmethod
+    def get_where_scoreboard_data(matchschedule_data):
+        where = "SG.GameId IN ("
+        for game in matchschedule_data:
+            if not game["MSFF"] or not game["MSGFF"]:
+                game_id = game["GameId"]
+                where += f'"{game_id}", '
+        where = where[:-2] + ")"
+        return where
+
+    def get_scoreboard_data(self, matchschedule_data):
+        where = self.get_where_scoreboard_data(matchschedule_data)
+        scoreboard_data = self.site.cargo_client.query(
+            tables="ScoreboardGames=SG, ScoreboardPlayers=SP",
+            fields=["SG.OverviewPage", "SG.Team1", "SG.Team2", "SP.IngameRole", "SP.Team", "SP.Link", "SG.GameId",
+                    "SG.MatchId"],
+            order_by="SG.N_Page, SG.N_MatchInPage, SG.N_GameInMatch",
+            where=where,
+            join_on="SG.GameId=SP.GameId"
+        )
+        return scoreboard_data
+
+    def process_matchschedule_data(self, matchschedule_data):
+        for match in matchschedule_data:
+            if not self.match_data.get(match["MatchId"]):
+                self.match_data[match["MatchId"]] = {"ff": False, "best_of": match["BestOf"], "team1": match["Team1"],
+                                                     "team2": match["Team2"], "games": {}}
+                if match["MSFF"]:
+                    self.match_data[match["MatchId"]]["ff"] = True
+                self.alt_teamnames[match["Team1"]] = match["Team1Final"]
+                self.alt_teamnames[match["Team2"]] = match["Team2Final"]
+            self.match_data[match["MatchId"]]["games"][match["GameId"]] = {"msg_data": match}
+
+    def get_player_id(self, player):
+        response = self.site.cargo_client.query(
+            tables="Players=P, PlayerRedirects=PR",
+            fields="P.Player",
+            where=f"PR.AllName = '{player}'",
+            join_on="P.OverviewPage=PR.OverviewPage"
+        )
+        if not response:
+            return player
+        return response[0]["Player"]
+
+    def process_scoreboard_data(self, scoreboard_data):
+        player_ids_cache = {}
+
+        for scoreboard in scoreboard_data:
+            if "sg_data" not in self.match_data[scoreboard["MatchId"]]["games"][scoreboard["GameId"]].keys():
```

Comment:
these are some long nesting dict keys, I might alias them as needed (and same thing in later methods)